rule.php

```diff
@@ -86,6 +122,18 @@ public static function validate_settings_form_fields(array $errors,
      *      which is the id of the quiz being saved.
      */
     public static function save_settings($quiz) {
+        $settingsmanager = new settings_manager();
+
+        $defaults = $settingsmanager->get_quiz_defaults();
+        // Set each required setting to default it not provided.
+        foreach ($defaults as $name => $defaultvalue) {
+            if (!isset($quiz->$name)) {
+                $quiz->$name = $defaultvalue;
+            }
+        }
+
+        // Save/update settings from form.
+        config_settings::with_form_data($quiz)->save_settings();
```

Comment:
Can see where w validate data. E.g. "seb_expressionsblocked" or "seb_expressionsallowed". Should we validate them? If so then create a card for that. We will treat it as a feature and do later.